File: Java Programs/160905318/week7/Stack.java
import java.util.LinkedList;
import java.util.Scanner;
interface Stack{
	void push(int n);
	int pop();
	void display();
}

class FixedStack implements Stack{
	int arr[],top,cap;
	FixedStack(int cap){
		this.cap=cap;
		top=-1;
		arr=new int[cap];
	}
	public void push(int n){
		if(top>=cap-1){
			System.out.println("Stack is full");
			return;
		}
		arr[++top]=n;
	}
	public int pop(){
		if(top<0){
			System.out.println("Stack empty");
			return Integer.MIN_VALUE;
		}
		return arr[top--];
	}
	public void display(){
		for(int i=0;i<=top;i++)
			System.out.print(arr[i]+" ");
		System.out.println();
	}
}

class DynamicStack implements Stack{
	LinkedList<Integer> list;
	DynamicStack(){
		list=new LinkedList<Integer>();
	}
	public void push(int n){
		list.addFirst(n);
	}
	public int pop(){
		if(list.size()==0){
			System.out.println("Stack empty");
			return Integer.MIN_VALUE;
		}
		return list.remove();
	}
	public void display(){
		Object arr[]=list.toArray();
		for(Object i:arr)
			System.out.print(((Integer)i)+" ");
		System.out.println();
	}
}

class TestStack{
	public static void main(String args[]){
		Scanner sc=new Scanner(System.in);
		Stack st;
		System.out.println("Enter 1 for fixed size and 2 for Dynamic Stack");
		int ch=sc.nextInt();
		if(ch==1){
			System.out.println("Enter capacity");
			st=new FixedStack(sc.nextInt());
		}
		else if(ch==2)
			st=new DynamicStack();
		else{
			System.out.println("Incorrect choice");
			return;
		}
		while(true){
			System.out.println("1. Push 2. Pop 3. Display 4. Exit");
			ch=sc.nextInt();
			switch (ch) {
				case 1: System.out.println("Enter element");
				st.push(sc.nextInt());
				break;
				case 2: int n=st.pop();
				if(n!=Integer.MIN_VALUE)
					System.out.println("Popped:"+n);
				break;
				case 3: st.display();
				break;
				case 4:return;
			}
		}
	}
}

File: DS - C Programs/Week4/stack.c
# include <stdio.h>
struct stack
{ char a[10];
  int top;
  };

 void push(char item,struct stack *s)
 { if(s->top==9)
   {printf("Stack is full");
   return;
   }
   else
   	{ printf("Push %c\n",item);
   	  s->top=s->top+1;
   	  s->a[s->top]=item; 
  }
}

  void pop(struct stack *s)
  { if(s->top==-1)
  	{printf("Stack is empty ");
    return;
    }
    else
    printf("Pop %c \n",s->a[s->top]);
     s->top=s->top-1; 
  }

  void display(struct stack *s)
  {int i=0;
  	printf("Stack is ");
   for(i=0;i<=s->top;i++)
   printf("%c \t",s->a[i]);
    }

 int main()
 { struct stack s;
 	char a;
   s.top=-1;
   int x=1;
   do
   { printf("\nEnter number 1-Push  2-Pop  3-Display 4-Exit ");
     scanf("%d",&x);
      if(x==1)
      	{ printf("Enter element ");
          scanf(" %c",&a);
           push(a,&s);
         }
      if(x==2)
         pop(&s);

      if(x==3)
      display(&s);

    }
   while(x==1 | x==2 | x==3);
   

   return 0;
}